src/sonic-yang-models/yang-models/sonic-portchannel.yang

```diff
@@ -95,7 +95,14 @@ module sonic-portchannel {
                     mandatory true;
                     type stypes:admin_status;
                 }
-            } /* end of list PORTCHANNEL_LIST */
+
+				leaf lacp_key {
+					type string {
+						/* lacp key should be either auto or a integer in the range of 1 to 65535 */
+						pattern '[1-9]{1,4}|[1-5][0-9]{4}|6[0-4][0-9]{3}|65[0-4][0-9]{2}|655[0-2][0-9]|6553[0-5]|auto';
+					}
```

Comment:
I feel we should keep "auto" string with default keyword -- to scenario where  user don't give any input

Reply:
We can't treat a blank string as "auto", if this is blank we are maintaining the old behavior of keeping lacp key as 0 (although it is not an allowed value by the LACP spec).

This is because we wanted to make sure that if a user is upgrading from an older version and he uses warm-reboot to do so, the connection will remain up and not lose any packets, and to do so we need to keep the lacp key as 0.
So, what we chose to do is to treat the scenrio of lacp_key empty\non existing as lacp key 0 and to not update the DB with lacp key when a warm update has occured.

Reply:
Ok I see,  Would be good to add a description on this default case as well then.